src/loki/reach.cc

```diff
@@ -125,20 +91,25 @@ directed_reach Reach::operator()(const DirectedEdge* edge,
     // expand from the node
     if (!reader.GetGraphTile(node_id, tile))
       continue;
+
     for (const auto& edge : tile->GetDirectedEdges(node_id)) {
       // get the opposing edge
       if (!reader.GetGraphTile(edge.endnode(), tile))
         continue;
-      const auto* node = tile->node(edge.endnode());
-      const auto* opp_edge = tile->directededge(node->edge_index() + edge.opp_index());
 
+      // get opposing edge id
+      GraphId opp_edgeid = edge.endnode();
+      opp_edgeid.set_id(tile->node(edge.endnode())->edge_index() + edge.opp_index());
+      // get opposing edge
+      const auto* opp_edge = tile->directededge(opp_edgeid);
```

Comment:
probably call the new helper function you made?

Reply:
we need to get `edge_id` at first to call new function. so, it will not simplify the code

Reply:
oh i see its because of the loop of edges doest give you the edge_id which the new function needs. well that is a shame. i suppose we could make a helper on `GraphTile::GetEdgeId(const DirectedEdge* e)` that would return the edge id if the edge belonged to the tile and throw otherwise but that would technically be more complex than this. :man_shrugging: nevermind